wagtail/users/views/groups.py

```diff
@@ -152,11 +159,21 @@ class GroupViewSet(ModelViewSet):
 
     @property
     def users_view(self):
-        return Index.as_view()
+        def view(request, pk):
+            legacy_url = reverse(self.get_url_name("users"), args=(pk,))
+            new_url = set_query_params(
+                reverse("wagtailusers_users:index"),
+                {"group": get_object_or_404(Group, pk=pk).pk},
+            )
 
-    @property
-    def users_results_view(self):
-        return Index.as_view(results_only=True)
+            warn(
+                f"Accessing the list of users in a group via {legacy_url} is "
+                f"deprecated, use {new_url} instead.",
+                RemovedInWagtail70Warning,
+            )
+            return redirect(new_url)
+
+        return view
```

Comment:
This makes the `GroupViewSet` not dependent of the users `IndexView` directly. However, one downside is that if you're viewing a group edit view, and someone else deletes that group, then you click on the "View users in this group" link, you'll get taken to an unfiltered users index view. This is because the filter param is now invalid, as the group no longer exists. (Unfortunately the filters popup button does not have the proper styling to indicate that the form has errors yet...)